common/utils/src/main/scala/org/apache/spark/util/MavenUtils.scala

```diff
@@ -405,6 +415,7 @@ private[spark] object MavenUtils extends Logging {
   def resolveMavenCoordinates(
       coordinates: String,
       ivySettings: IvySettings,
+      noCacheIvySettings: Option[IvySettings] = None,
```

Comment:
Where was a non-None `noCacheIvySettings` passed in?

Reply:
I added the input for `noCacheIvySettings` in `IsolatedClientLoader` and built a new client. Although the retry was successful, there will be a large number of error messages like the following:

```
:: problems summary ::
:::: ERRORS
	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache

	unknown resolver local-m2-cache


```

Reply:
> Where was a non-None `noCacheIvySettings` passed in? I build a new client with this pr , but also found the error message:
> 
> ```
> :: problems summary ::
> :::: WARNINGS
> 		[NOT FOUND  ] org.apache.hive#hive-exec;2.3.9!hive-exec.jar (0ms)
> 
> 	==== local-m2-cache: tried
> 
> 	  file:/Users/yangjie01/.m2/repository/org/apache/hive/hive-exec/2.3.9/hive-exec-2.3.9.jar
> 
> 		[NOT FOUND  ] asm#asm;3.2!asm.jar (1ms)
> 
> 	==== local-m2-cache: tried
> 
> 	  file:/Users/yangjie01/.m2/repository/asm/asm/3.2/asm-3.2.jar
> 
> 		::::::::::::::::::::::::::::::::::::::::::::::
> 
> 		::              FAILED DOWNLOADS            ::
> 
> 		:: ^ see resolution messages for details  ^ ::
> 
> 		::::::::::::::::::::::::::::::::::::::::::::::
> 
> 		:: asm#asm;3.2!asm.jar
> 
> 		:: org.apache.hive#hive-exec;2.3.9!hive-exec.jar
> 
> 		::::::::::::::::::::::::::::::::::::::::::::::
> ```

Yes, this is my bad, it should be passed in when `IsolatedClientLoader` calls it.

Reply:
> I added the input for `noCacheIvySettings` in `IsolatedClientLoader` and built a new client. Although the retry was successful, there will be a large number of error messages like the following:
> 
> ```
> :: problems summary ::
> :::: ERRORS
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> 
> 	unknown resolver local-m2-cache
> ```

The above issues have been resolved.